pkg/azure/provider/helpers/providerspec_test.go

```diff
@@ -0,0 +1,110 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package helpers
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/gardener/machine-controller-manager-provider-azure/pkg/azure/api"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/ptr"
+)
+
+func TestNilConfig(t *testing.T) {
+	g := NewWithT(t)
+
+	var (
+		testConfig *api.CloudConfiguration = nil
+		testRegion                         = ptr.To("Foo")
+	)
+
+	configuration, err := DetermineCloudConfiguration(testConfig, testRegion)
+
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(configuration).To(Equal(cloud.AzurePublic))
+}
+
+func TestNilRegion(t *testing.T) {
+	g := NewWithT(t)
+
+	var (
+		testConfig         = &api.CloudConfiguration{Name: api.AzurePublicCloudName}
```

Comment:
You should pick some other cloud name since the default is also `AzurePublic`. This will ensure that this test will not pass if the code under test has been changed incorrectly.